File: VB/InfoPanels.Module.Web/Controllers/InfoPanelViewControllerWeb.vb
Imports System.ComponentModel
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.SystemModule
Imports System.Web.UI
Imports DevExpress.ExpressApp.Templates

Namespace InfoPanels.Module.Web.Controllers
    Public Interface IInfoPanelTemplateWeb
        Inherits IFrameTemplate

        ReadOnly Property PlaceHolder() As Control
    End Interface
    Public Class InfoPanelViewControllerWeb
        Inherits CustomizeTemplateViewControllerBase(Of IInfoPanelTemplateWeb)

        Private literal As LiteralControl
        Protected Overrides Sub AddControlsToTemplateCore(ByVal template As IInfoPanelTemplateWeb)
            If literal Is Nothing Then
                literal = New LiteralControl()
            End If
            If template.PlaceHolder IsNot Nothing Then
                template.PlaceHolder.Controls.Add(literal)
            End If
        End Sub
        Protected Overrides Sub RemoveControlsFromTemplateCore(ByVal template As IInfoPanelTemplateWeb)
            If template.PlaceHolder IsNot Nothing Then
                template.PlaceHolder.Controls.Remove(literal)
                literal = Nothing
            End If
        End Sub
        Protected Overrides Sub UpdateControls(ByVal view As View)
            UpdateControlsCore()
        End Sub
        Protected Overrides Sub UpdateControls(ByVal currentObject As Object)
            UpdateControlsCore()
        End Sub
        Private Sub UpdateControlsCore()
            literal.Text = "The current View is " & View.Caption
            If View.CurrentObject IsNot Nothing Then
                literal.Text &= "<br/>The current object is " & View.CurrentObject.ToString()
            End If
        End Sub
    End Class
End Namespace

File: CS/InfoPanels.Module.Web/Controllers/InfoPanelViewControllerWeb.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.SystemModule;
using System.Web.UI;
using DevExpress.ExpressApp.Templates;

namespace InfoPanels.Module.Web.Controllers {
    public interface IInfoPanelTemplateWeb : IFrameTemplate {
        Control PlaceHolder {
            get;
        }
    }
    public class InfoPanelViewControllerWeb : CustomizeTemplateViewControllerBase<IInfoPanelTemplateWeb> {
        LiteralControl literal;
        protected override void AddControlsToTemplateCore(IInfoPanelTemplateWeb template) {
            if (literal == null) literal = new LiteralControl();
            if (template.PlaceHolder != null) {
                template.PlaceHolder.Controls.Add(literal);
            }
        }
        protected override void RemoveControlsFromTemplateCore(IInfoPanelTemplateWeb template) {
            if (template.PlaceHolder != null) {
                template.PlaceHolder.Controls.Remove(literal);
                literal = null;
            }
        }
        protected override void UpdateControls(View view) {
            UpdateControls();
        }
        protected override void UpdateControls(object currentObject) {
            UpdateControls();
        }
        void UpdateControls() {
            literal.Text = "The current View is " + View.Caption;
            if (View.CurrentObject != null) {
                literal.Text += "<br/>The current object is " + View.CurrentObject;
            }
        }
    }
}
